src/store/index.js

```diff
@@ -38,14 +47,37 @@ const eventsSlice = createSlice({
   reducers: {},
   extraReducers: {
     [fetchIndexData.pending]: (state, action) => {
-      state.status = FETCH_INDEX_PAGE_INFO_STATUS.LOADING
+      const reset = action.meta.arg.reset
+      if (reset) {
+        state.status = FETCH_INDEX_PAGE_INFO_STATUS.LOADING
+      } else {
+        state.status = FETCH_INDEX_PAGE_INFO_STATUS.LOADING_MORE
+      }
     },
     [fetchIndexData.fulfilled]: (state, action) => {
-      const { poapEvents, mostRecent, mostClaimed, upcoming } = action.payload
-      state.events = poapEvents
-      state.mostRecent = mostRecent
-      state.mostClaimed = mostClaimed
-      state.upcoming = upcoming
+      const { poapEvents, apiSkip, mainnetSkip, xdaiSkip, page, total, invalid } = action.payload
+
+      if (page === 0) {
+        state.currentInvalidResults = invalid
+        state.events = poapEvents
+      } else {
+        state.currentInvalidResults += invalid
+        poapEvents.forEach(poapE => {
+          const match = state.events.find(e => e.id === poapE.id)
+          if (match) {
+            match.tokenCount += poapE.tokenCount
+            match.transferCount += poapE.transferCount
```

Comment:
Aren't we doing a mutation here?

Reply:
If event is already loaded instead of adding it we aggregate the data